File: FrontendApp/src/app/models/dtos/invitation.dto.ts
import { CompanyRole } from "../../enums/company-role.enum";
import { InvitationType } from "../../enums/invitation-type.enum";

export interface InvitationDTO {
  id: number;
  description: string | null;
  companyId: number | null;
  role: CompanyRole | null;
  teamId: number | null;
  type: InvitationType | null;
}

File: Services/UnionService/UnionService.Application/DTOs/InvitationDTO.cs
using UnionService.Domain.Enums;

namespace UnionService.Application.DTOs
{
    public class InvitationDTO
    {
        public int Id { get; set; }
        public string? Description { get; set; }
        public int UserId { get; set; }
        public int CompanyId { get; set; }
        public CompanyRole? Role { get; set; }
        public int? TeamId { get; set; }
        public InvitationType Type { get; set; }
    }
}
